src/anchor-navigation/scroll-spy.tsx

```diff
@@ -0,0 +1,95 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { useCallback, useEffect, useState } from 'react';
+
+const isBrowser = typeof window !== 'undefined';
+
+export default function useScrollSpy({
+  hrefs,
+}: {
+  hrefs: string[];
+}): [
+  string | undefined,
+  React.Dispatch<React.SetStateAction<string | undefined>>,
+  React.Dispatch<React.SetStateAction<boolean>>
+] {
+  const [currentHref, setCurrentHref] = useState<string>();
+  const [scrollY, setScrollY] = useState(isBrowser ? window.pageYOffset : 0);
```

Comment:
Because `scrollY` is a state variable, every time the user scrolls the component will re-render, even if it does not need to. Is there a way we can optimize this?